.evergreen/build_all.sh

```diff
@@ -76,6 +78,7 @@ echo "Installing libmongocrypt"
 $CMAKE --build . --target install --config RelWithDebInfo
 $CMAKE --build . --target test-mongocrypt --config RelWithDebInfo
 $CMAKE --build . --target test_kms_request --config RelWithDebInfo
+"$CTEST" -C RelWithDebInfo
```

Comment:
The test logs are now a [summary](https://evergreen.mongodb.com/lobster/evergreen/task/libmongocrypt_rhel_62_64_bit_build_and_test_and_upload_patch_b12cb018bbd1a4e8a7fb69f8baf241daf3f0e939_62cc5c139ccd4e4e1b0669bd_22_07_11_17_21_23/0/task#bookmarks=0%2C5735&l=1&shareLine=5382).

Adding a test failure locally does not output logs by default:
```
$ ctest .
(omitted for brevity)
The following tests FAILED:
          1 - mongocrypt (Subprocess aborted)
Errors while running CTest
Output from these tests are in: /Users/kevin.albertson/review/libmongocrypt-413/cmake-build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```

Using the suggested options, the output of stderr is output on one line:
```
$ ctest . --output-on-failure --rerun-failed
(omitted for brevity)
No 'real' csfle library is available. The _test_encrypt_csfle_no_needs_markings test is a no-op.No 'real' csfle library is available. The _test_encrypt_fle2_explain_with_csfle test is a no-op.No 'real' csfle library is available. The _test_encrypt_fle1_explain_with_csfle test is a no-op.No 'real' csfle library is available. The _test_fle1_create_with_csfle test is a no-op.test error /Users/kevin.albertson/review/libmongocrypt-413/test/test-mongocrypt-ctx-encrypt.c:4281 _test_fle2_create(): !mongocrypt_ctx_finalize (ctx, out) failed with msg: (null)
(omitted for brevity)
0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.13 sec

The following tests FAILED:
          1 - mongocrypt (Subprocess aborted)
Errors while running CTest
```

If a test fails on Evergreen, I prefer it be easy to identify the failing test function and the failing assertion. Can the stderr and stdout logs be printed on failure in an easily readable form?

Reply:
`--output-on-failure` should be added, yes. I forget that its not the default because I use an environment variable to turn that on locally.

The repeated `No 'real' csfle library is available. The _test_encrypt_csfle_no_needs_markings test is a no-op.` on a single line is actually just a bug (the message is missing a newline character).